dispatcher/dispatcher.go

```diff
@@ -59,7 +67,7 @@ func (d dispatcher) Do(ctx context.Context, maxIter int, callback func()) error
 	}
 
 	wg.Wait()
-	return nil
```

Comment:
[question]
Why were we always returning a nil error before? :thinking:

Reply:
Ok, thanks, I was surprised we returned anything in this case, but we're not doing it anymore anyways!